secure-connection/register.go

```diff
@@ -17,15 +17,27 @@ func register(err error, client *nex.Client, callID uint32, stationUrls []*nex.S
 	client.SetConnectionID(nextConnectionID)
 
 	localStation := stationUrls[0]
-	publicStation := localStation.Copy()
 
-	publicStation.SetAddress(client.Address().IP.String())
-	publicStation.SetPort(uint32(client.Address().Port))
-	publicStation.SetNatf(0)
-	publicStation.SetNatm(0)
-	publicStation.SetType(3)
+	// * A NEX client can set the public station URL by setting two URLs on the array
+	var publicStation *nex.StationURL
+	if len(stationUrls) > 1 {
+		publicStation = stationUrls[1]
+	} else {
```

Comment:
Public stations seem to use a `type` parameter of 3. This feels like a safer way to check for the public station rather than assuming it's the 2nd value in the list

Out of curiosity, how does it _know_ it's public station here?

Reply:
> Out of curiosity, how does it know it's public station here?

I'm not sure tbh. It seems to be a thing external to the server though, because it also uses the public IP even when the server is on the same network as the client